Ext: Add showInfos and tooltips to the forms (#1817)
kyma-project/busola: docs/extensibility/form-section.md

Text widgets render a field as a text field. They are used by default for all string values.

#### Widget-specific parameters

- **enum[]** - an array of options to generate an input field with a dropdown. Optionally can be a string containing a JSONata expression returning an array of options.
- **placeholder** - specifies a short hint about the input field value.
- **required** - a boolean which specifies if a field is required. The default value is taken from CustomResourceDefintion (CRD); if it doesn't exist in the CRD, then it defaults to `false`.

> - **inputInfo** - a string that can be shown below the input field to guide the User how to fill in the input.

Reviewer: ```suggestion
- **inputInfo** - a string below the input field that shows how to fill in the input.
```

#### Widget-specific parameters

- **extraPaths** - an array of extra paths to fill in with the contents of the field. Each path can either be a period-separated string or an array of strings.
- **placeholder** - specifies a short hint about the input field value.

> - **inputInfo** - a string that can be shown below the input field to guide the User how to fill in the input. It defaults to `Name must consist of lowercase alphanumeric characters, can contain '-' and '.' (e.g.: 'my.name-1').`. To disable any suggestion, set this value to `null`.

Reviewer: ```suggestion
- **inputInfo** - a string below the input field that shows how to fill in the input. It defaults to `Name must consist of lowercase alphanumeric characters, can contain '-' and '.' (e.g.: 'my.name-1').`. To disable any suggestion, set this value to `null`.
```

> - **description** - a string that will be displayed in a tooltip when hovering over a question mark icon, next to the input's label. The default value is taken from the CustomResourceDefintion (CRD).

Reviewer: ```suggestion
- **description** - a string displayed in a tooltip when you hover over a question mark icon, next to the input's label. The default value is taken from the CustomResourceDefintion (CRD).
```

CodeEditor widgets render a versatile code editor that can be used to edit any variable. The editor's default language is JSON.

#### Widget-specific parameters

> - **inputInfo** - a string that can be shown below the input field to guide the User how to fill in the input.

Reviewer: ```suggestion
- **inputInfo** - a string below the input field that shows how to fill in the input.
```

> - **description** - a string that will be displayed in a tooltip when hovering over a question mark icon, next to the input's label. The default value is taken from the CustomResourceDefintion (CRD).

Reviewer: ```suggestion
- **description** - a string displayed in a tooltip when you hover over a question mark icon, next to the input's label. The default value is taken from the CustomResourceDefintion (CRD).
```

  - **scope** - either `namespace` or `cluster`. When set to `cluster`, namespaced resources are fetched from all Namespaces. Defaults to `cluster`.
  - **namespace** - Namespace to fetch resources from. Used only when scope is `namespace` and resources need to be fetched from a specific Namespace. Defaults to the active Namespace when omitted.
- **required** - a boolean which specifies if a field is required. The default value is taken from CRD; if it doesn't exist in CRD, then it defaults to `false`.

> - **inputInfo** - a string that can be shown below the input field to guide the User how to fill in the input.

Reviewer: ```suggestion
- **inputInfo** - a string below the input field that shows how to fill in the input.
```

> - **description** - a string that will be displayed in a tooltip when hovering over a question mark icon, next to the input's label. The default value is taken from the CustomResourceDefintion (CRD).

Reviewer: ```suggestion
- **description** - a string displayed in a tooltip when you hover over a question mark icon, next to the input's label. The default value is taken from the CustomResourceDefintion (CRD).
```

#### Widget-specific parameters

- **required** - a boolean which specifies if a field is required. The default value is taken from CRD; if it doesn't exist in the CRD, then it defaults to `false`.
- **keyEnum[]** - an array of options to generate a key input field with a dropdown.
- **value**:
  - **type** - a string that specifies the type of the value input. The options are `object`, `number`, `text`. Defaults to `text`.
  - **keyEnum[]** - an array of options to generate a key input field with a dropdown only if the `type` is set to `object`.
  - **valueEnum[]** - an array of options to generate a value input field with a dropdown.

> - **inputInfo** - a string that can be shown below the last input field to guide the User how to fill in the input.

Reviewer: ```suggestion
- **inputInfo** - a string below the input field that shows how to fill in the input.
```

> - **description** - a string that will be displayed in a tooltip when hovering over a question mark icon, next to the input's label. The default value is taken from the CustomResourceDefintion (CRD).

Reviewer: ```suggestion
- **description** - a string displayed in a tooltip when you hover over a question mark icon, next to the input's label. The default value is taken from the CustomResourceDefintion (CRD).
```

<img src="./assets/form-widgets/GenericList.png" alt="Example of a GenericList widget" style="border: 1px solid #D2D5D9">

### SimpleList

SimpleList widgets render an `array` as a table with rows representing data items and columns representing different fields. New items are added automatically when new entries are typed in.

This type of field is only suitable for simple data types and can contain more complex structures in its items.

#### Widget-specific parameters

- **placeholder** - specifies a short hint about the input field value.
- **required** - a boolean which specifies if a field is required. The default value is taken from CRD; if it doesn't exist in the CRD, then it defaults to `false`.

> - **inputInfo** - a string that can be shown below the last input field to guide the User how to fill in the input.

Reviewer: ```suggestion
- **inputInfo** - a string below the input field that shows how to fill in the input.

```

> - **description** - a string that will be displayed in a tooltip when hovering over a question mark icon, next to the input's label. The default value is taken from the CustomResourceDefintion (CRD).

Reviewer: ```suggestion
- **description** - a string displayed in a tooltip when you hover over a question mark icon, next to the input's label. The default value is taken from the CustomResourceDefintion (CRD).
```
